lightning/src/routing/router.rs

```diff
@@ -1442,24 +1442,31 @@ where L::Target: Logger {
 	}
 
 	// Sort by total fees and take the best paths.
-	payment_paths.sort_by_key(|path| path.get_total_fee_paid_msat());
+	payment_paths.sort_by_cached_key(|path| path.get_total_fee_paid_msat());
```

Comment:
Should we just swap for `sort_unstable_by_key` instead? I kinda doubt the addition is all that slow, given most routes are only 2-5 hops in length and it avoids temporary allocation.

Reply:
Hm, in order to get a first impression of the performance at this point, I added a [test output for the different `sort` variants](https://gist.github.com/tnull/86ef4f5e87b40d0ce829196e2010dfd3) just above the marked line and ran the `router::tests`.  [The results](https://gist.github.com/tnull/d874da6a0dcac2181e57f4b01a325151) are kind of mixed, with `sort_by_cached_key` taking twice as long in some instances and speeding up the sorting more than tenfold in others. My assumption however would be that for real world scenarios (where  `payment_paths` would be larger), `sort_by_cached_key` would tend to further outperform `sort_unstable_by_key`?

That said, I have no strong opinion on this and if you're wary of the memory access, I'm happy to change it to `sort_unstable_by_key`.

Reply:
Hmm, I could be wrong, but I'd assume pretty strongly that these sorts are fast enough that they're totally invisible compared to the actual graph walk, and in that context skipping the allocation is nice. Its more of a minor point here, though.

Reply:
Alright, then `s/sort_by_cached_key`/`sort_unstable_by_key` is fine by me.

Reply:
Done with https://github.com/lightningdevkit/rust-lightning/pull/1359/commits/3c69c9133b1a744de9001d9276d060d5f094b298.